crowbar_framework/app/controllers/nodes_controller.rb

```diff
@@ -106,13 +106,8 @@ def bulk
             dirty = false
             node = NodeObject.find_node_by_name node_name
 
-            unless node.alias == node_attributes["alias"]
-              node.force_alias = node_attributes["alias"]
-              dirty = true
-            end
-
-            unless node.public_name == node_attributes["public_name"]
-              node.force_public_name = node_attributes["public_name"]
+            if node_attributes["allocate"] and not node.allocated?
+              node.allocate!
               dirty = true
```

Comment:
You need to save `node.allocated?` before allocating the node, as it's used later on :-)